extensions/amp-iframe/0.1/amp-iframe.js

```diff
@@ -531,6 +541,80 @@ export class AmpIframe extends AMP.BaseElement {
     }
   }
 
+  /**
+   * Requests consent data from consent module
+   * and forwards information to iframe
+   * @param {Window} source
+   * @param {string} origin
+   * @private
+   */
+  sendConsentData_(source, origin) {
+    const consentPolicyId = super.getConsentPolicy() || 'default';
+    const consentStringPromise = this.getConsentString_(consentPolicyId);
+    const metadataPromise = this.getConsentMetadata_(consentPolicyId);
+    const consentPolicyStatePromise = this.getConsentPolicyState_(
+      consentPolicyId
+    );
+
+    Promise.all([
+      metadataPromise,
+      consentStringPromise,
+      consentPolicyStatePromise,
+    ]).then((consents) => {
+      this.sendConsentDataToIframe_(
+        source,
+        origin,
+        dict({
+          'type': MessageType.CONSENT_DATA,
+          'consentMetadata': consents[0],
+          'consentString': consents[1],
```

Comment:
How do we usually define data types for inbound messages? I wonder if it would make sense to have an intermediate `consent` object with these values.

Reply:
What I've found: for **embed-size** the properties `height` and `width` are next to `type` (see https://github.com/ampproject/amphtml/blob/master/extensions/amp-iframe/0.1/amp-iframe.js#L467-L476). **intersection** event sends `changes` next to `type` (see https://github.com/ampproject/amphtml/blob/master/src/utils/intersection-observer-3p-host.js#L117-L120).